packages/@react-spectrum/card/docs/CardView.mdx

```diff
@@ -0,0 +1,507 @@
+in the layout calculations.
+
+```tsx example
+// Using the same CardView as above the GridLayout example, but applying a modified GalleryLayout instead
+function CustomGalleryLayout() {
+  let {scale} = useProvider();
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let galleryLayout = React.useMemo(() =>
+    new GalleryLayout({
+      collator,
+      scale,
+      itemSpacing: new Size(50, 18)
+    })
+  , [collator, scale]);
+
+  return (
+    <CustomLayout layout={galleryLayout} aria-label="CardView with modified GalleryLayout" />
+  );
+}
+
+<CustomGalleryLayout />
+```
+
+
+### WaterfallLayout
+The WaterfallLayout organizes the cards into equal width columns but allows each card to have a variable height. Similarly to GalleryLayout, it attempts to accommodate
+the aspect ratio of each card's image. The customizable options available to the GalleryLayout are as follows:
+
+// TODO figure out how to show a table for the WaterfallLayoutOptions
+
+The example below shows a CardView with a modified WaterfallLayout adjusts the minimum required space between the cards and sets the maximum number of columns to 2.
+
+```tsx example
+// Using the same CardView as above the GridLayout example, but applying a modified WaterfallLayout instead
+function CustomWaterfallLayout() {
+  let {scale} = useProvider();
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let waterfallLayout = React.useMemo(() =>
+    new WaterfallLayout({
+      collator,
+      scale,
+      maxColumns: 2,
+      minSpace: new Size(50, 50)
+    })
+  , [collator, scale]);
+
+  return (
+    <CustomLayout layout={waterfallLayout} aria-label="CardView with modified WaterfallLayout" />
+  );
+}
+
+<CustomWaterfallLayout />
+```
+
+## Labeling
+### Accessibility
+An `aria-label` must be provided to the CardView for accessibility. If the CardView is labeled by a separate element, an `aria-labelledby` prop must be provided using the id of the labeling element instead.
+
+## Asynchronous loading
+CardView supports loading data asynchronously, and will display a progress circle reflecting the current load state,
+set by the `loadingState` prop. It also supports infinite scrolling to load more data on demand as the user scrolls, via the `onLoadMore` prop.
+
+This example uses the [useAsyncList](../react-stately/useAsyncList.html) hook to handle loading the data.
+See the docs for more information.
+
+```tsx example
+import {useAsyncList} from '@react-stately/data';
+
+function AsyncLoadingCardView() {
+  let imageLinks = React.useRef(null);
+  let list = useAsyncList({
+    async load({signal, cursor}) {
+      if (cursor) {
+        cursor = cursor.replace(/^http:\/\//i, 'https://');
+      }
+
+      if (imageLinks.current == null) {
+        let fullRes = await fetch('https://rawcdn.githack.com/akabab/starwars-api/0.2.1/api/all.json');
+        imageLinks.current = await fullRes.json();
+      }
```

Comment:
A bit scuffed since I had to source the image urls from another api that just returns all the results in one call